server/plugin/utils_test.go

```diff
@@ -615,3 +617,87 @@ func TestGetConnectAccountFirstMessage(t *testing.T) {
 		})
 	}
 }
+
+func TestGetOffsetAndLimitFromQueryParams(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+	for _, testCase := range []struct {
+		description       string
+		queryParamPage    string
+		queryParamPerPage string
+		expectedOffset    int
+		expectedLimit     int
+	}{
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: valid page and per_page query params",
+			queryParamPage:    "1",
+			queryParamPerPage: "10",
+			expectedOffset:    10,
+			expectedLimit:     10,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: empty page and per_page query params",
+			queryParamPage:    "",
+			queryParamPerPage: "",
+			expectedOffset:    0,
+			expectedLimit:     constants.DefaultPerPageLimit,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: invalid page query param",
+			queryParamPage:    "invalidNonIntegerString",
+			queryParamPerPage: "10",
+			expectedOffset:    0,
+			expectedLimit:     10,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: invalid per_page query param",
+			queryParamPage:    "1",
+			queryParamPerPage: "invalidNonIntegerString",
+			expectedOffset:    constants.DefaultPerPageLimit,
+			expectedLimit:     constants.DefaultPerPageLimit,
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
```

Comment:
Use something like this - https://github.com/Brightscout/mattermost-plugin-servicenow/blob/ae96b8fd1ae5a1da71575742894cd801fa55d3fa/server/testutils/mock.go#L7